hera_qm/utils.py

```diff
@@ -138,3 +140,79 @@ def generate_fullpol_file_list(files, pol_list):
                 file_list.append(jd_list)
 
     return file_list
+
+
+def metrics2mc(filename, ftype):
+    """ Read in file containing quality metrics and stuff into a dictionary which
+    can be used by M&C to populate db.
+
+    Args:
+        filename: (str) file to read and convert
+        ftype: (str) Type of metrics file. Options are ['ant', 'firstcal', 'omnical']
+    Returns:
+        d: (dict) Dictionary containing keys and data to pass to M&C.
+            Structure is as follows:
+            d['ant_metrics']: Dictionary with metric names
+                d['ant_metrics'][metric]: list of lists, each containing [ant, pol, val]
+            d['array_metrics']: Dictionary with metric names
+                d['array_metrics'][metric]: Single metric value
+    """
+    d = {'ant_metrics': {}, 'array_metrics': {}}
+    if ftype is 'ant':
+        from hera_qm.ant_metrics import load_antenna_metrics
+        data = load_antenna_metrics(filename)
+        key2cat = {'final_metrics': 'ant_metrics',
+                   'final_mod_z_scores': 'ant_metrics_mod_z_scores'}
+        for key, category in key2cat.items():
+            for met, array in data[key].items():
+                metric = '_'.join([category, met])
+                d['ant_metrics'][metric] = []
+                for antpol, val in array.items():
+                    d['ant_metrics'][metric].append([antpol[0], antpol[1], val])
+            for met in ['crossed_ants', 'dead_ants', 'xants']:
+                metric = '_'.join(['ant_metrics', met])
+                d['ant_metrics'][metric] = []
+                for antpol in data[met]:
+                    d['ant_metrics'][metric].append([antpol[0], antpol[1], 1])
+            d['ant_metrics']['ant_metrics_removal_iteration'] = []
+            metric = 'ant_metrics_removal_iteration'
+            for antpol, val in data['removal_iteration'].items():
+                d['ant_metrics'][metric].append([antpol[0], antpol[1], val])
+
+    elif ftype is 'firstcal':
+        from hera_qm.firstcal_metrics import load_firstcal_metrics
+        data = load_firstcal_metrics(filename)
+        pol = data['pol']
+        d['array_metrics']['firstcal_metrics_good_sol'] = float(data['good_sol'])
```

Comment:
Actually, this gets recast in M&C now, so the float part here isn't necessary. I will take it out.